qemu/tests/blockdev_commit_query_named_block_nodes.py

```diff
@@ -0,0 +1,40 @@
+from provider import backup_utils
+from provider import job_utils
+
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitQueryNamedBlockNodes(BlockDevCommitTest):
+
+    def commit_snapshots(self):
+        device = self.params.get("device_tag")
+        device_params = self.params.object_params(device)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        self.device_node = self.get_node_name(device)
+        options = ["base-node", "top-node", "speed"]
+        arguments = self.params.copy_from_keys(options)
+        arguments["base-node"] = self.get_node_name(device)
+        arguments["top-node"] = self.get_node_name(snapshot_tags[-2])
+        device = self.get_node_name(snapshot_tags[-1])
+        commit_cmd = backup_utils.block_commit_qmp_cmd
+        cmd, args = commit_cmd(device, **arguments)
+        self.main_vm.monitor.cmd(cmd, args)
+        job_id = args.get("job-id", device)
+        if self.params.get("query_named_block_nodes", "no"):
```

Comment:
we do the query block nodes test in the case, so we may not need a flag here

Reply:
Have updated.